src/components/MainApp.js

```diff
@@ -7,42 +7,56 @@ import useData from "../useData";
 
 const ALIAS_MATERIAS = require("../data/materias.json");
 
+const allCodigos = new Set(
+  Object.keys(ALIAS_MATERIAS).map((c) => c.toLowerCase()),
+);
+
 const MainApp = () => {
   const { data, partialLoading } = useData();
   const repos = React.useMemo(() => {
     return data.map((r) => ({
       user: r.owner.login,
       repoName: r.name,
       description: r.description,
+      codigos: new Set(r.topics).intersection(allCodigos),
       repoData: r,
     }));
   }, [data]);
 
-  const materias = React.useMemo(() => {
-    const mapa = data.reduce((mapa, repo) => {
-      const codigosEnRepo = repo.topics
-        .map((t) => t.toUpperCase())
-        .filter((t) => ALIAS_MATERIAS.hasOwnProperty(t));
-      codigosEnRepo.forEach((codigoEnRepo) => {
-        const nombreMateria = ALIAS_MATERIAS[codigoEnRepo];
-        const valuesAntes = mapa.get(nombreMateria);
-        const codigosAntes = valuesAntes ? valuesAntes.codigos : [];
-        const reposAntes = valuesAntes ? valuesAntes.reponames : [];
-        mapa.set(nombreMateria, {
-          codigos: [...new Set([...codigosAntes, codigoEnRepo])],
-          reponames: [...new Set([...reposAntes, repo.full_name])],
+  // Es un Map<string, {codigos, reponames}>
+  // Esto permite que se muestren todos los codigos de una materia con un mismo nombre
+  const materiasPorNombre = React.useMemo(() => {
+    const materiasPorNombre = new Map();
+    for (const [key, value] of Object.entries(ALIAS_MATERIAS)) {
```

Comment:
```suggestion
    for (const [codigo, nombreMateria] of Object.entries(ALIAS_MATERIAS)) {
```

Algo para aclarar que es key y value?

Reply:
Se me pasó por alto 👍

Reply:
Creo que luego voy a tener que cambiar el codigo para adaptarlos con esos nombres